### PR TITLE
feat(router): Add new JWT authentication variants and use them

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Added merchant create routes
- Add dashboard routes
- add user creation routes
- Support user based JWT auth for routes

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

To make some routes accessible with JWT authentication

## How did you test it?

Postman.

### Test Cases
1. New APIs
```
curl --location 'http://localhost:8080/user/v2/signin' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "email",
    "password": "password"
}'
```
```
curl --location 'http://localhost:8080/user/v2/signup' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "email",
    "password": "password"
}'
```
```
curl --location 'http://localhost:8080/user/signin' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "email",
    "password": "password"
}'
```
```
curl --location 'http://localhost:8080/user/signup' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "email",
    "password": "password"
}'
```
All the above APIs will give the following response
```
{
    "token": "JWT Token",
    "merchant_id": "merchant_id",
    "name": "user name",
    "email": "email",
    "verification_days_left": null,
    "user_role": "org_admin"
}
```

2. Old APIs
The following APIs will now work with jwt tokens in the authorization header.
- Steps:
a. Get the JWT Token from the routes mentioned above
b. Put that token in the `Bearer Token` in `Authorization` tab of postman request
c. Hit the API, then it should work.

- APIs:
Merchant Account - Retrieve
Merchant Account - Update
Payment Connector - Create
Payment Connector - Retrieve
Payment Connector - List
Payment Connector - Update
Payment Connector - Delete
Business Profile - Create
Business Profile - Retrieve
Business Profile - Update
Business Profile - List
Api Key - Create
Api Key - Retrieve
Api Key - Revoke
Api Key - List
Payments - Create
Payments - Retrieve
Payments - List
Refunds - Create
Refunds - Retrieve
Refunds - List

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by lsampras — APPROVED** [review]
LGTM

**Review by apoorvdixit88 — COMMENTED** [review]
LGTM
(We can discuss whether we want to keep the blocked_email list or we can totally remove it from signup flow)

File: crates/router/src/utils/user/password.rs
Comment (SanchithHegde) [inline crates/router/src/utils/user/password.rs:1]: Can we have unit tests for these? Can be added in a separate PR, not a problem.

**Review by SanchithHegde — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

File: crates/router/src/consts.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#[cfg(feature = "olap")]
+pub mod user;
```
[the remark this answers is not in the document]
Comment (ThisIsMani) [reply crates/router/src/consts.rs:2]: Once we add new APIs and new roles, this file is going to become big.

File: crates/router/src/utils/user/password.rs
Comment (ThisIsMani) [reply crates/router/src/utils/user/password.rs:1]: Sure, will try to add them in a separate PR.

**Pull request merged**